backends/candle/src/models/modernbert.rs

```diff
@@ -577,28 +577,49 @@ impl ModernBertModel {
         Ok(extended_attention_mask)
     }
 
-    fn get_local_attention_mask(&self, attention_mask: &Tensor) -> Result<Tensor> {
-        let attention_mask = attention_mask.to_dtype(DType::U8)?;
+    fn get_window_mask(&self, seq_len: usize) -> Result<Tensor> {
+        let inverted_window_mask: Vec<f32> = (0..seq_len)
+            .flat_map(|i| {
+                (0..seq_len).map(move |j| {
+                    if (j as i32 - i as i32).abs() > self.window_size as i32 {
+                        1.
+                    } else {
+                        0.
+                    }
+                })
+            })
+            .collect();
+
+        let inverted_window_mask =
+            Tensor::from_slice(&inverted_window_mask, (seq_len, seq_len), &self.device)?;
+
+        Ok(inverted_window_mask)
+    }
 
-        let mask_shape = attention_mask.shape();
-        let (_, _, seq_len, _) = mask_shape.dims4()?;
+    fn get_attention_mask(
+        &self,
+        attention_mask: Option<&Tensor>,
+        input_shape: &(usize, usize),
+    ) -> Result<(Tensor, Tensor)> {
+        let global_attention_mask = self
+            .get_global_attention_mask(attention_mask, input_shape)?
+            .to_dtype(self.dtype)?;
 
-        let rows = Tensor::arange(0, seq_len as i64, attention_mask.device())?.unsqueeze(0)?;
-        let rows = rows.broadcast_as((seq_len, seq_len))?;
+        let min_value = match self.dtype {
+            DType::F32 => f32::MIN as f64,
+            _ => -65504.0, // f16 minimum value
+        };
 
-        let distance = (&rows - &rows.t()?)?.abs()?;
+        let global_attention_mask = ((1.0 - global_attention_mask)? * min_value)?;
+        let global_attention_mask = global_attention_mask.to_dtype(self.dtype)?;
 
-        let window_size = (self.local_attention / 2) as i64;
-        let window_mask = distance
-            .le(window_size)?
-            .unsqueeze(0)?
-            .unsqueeze(0)?
-            .broadcast_as(mask_shape)?;
+        let seq_len = global_attention_mask.dim(2)?;
+        let window_mask = self.get_window_mask(seq_len)?;
+        let window_mask = (window_mask * min_value)?.to_dtype(self.dtype)?;
 
-        let zero_tensor = Tensor::zeros_like(&attention_mask)?;
-        let local_attention_mask = attention_mask.where_cond(&window_mask, &zero_tensor)?;
+        let local_attention_mask = global_attention_mask.broadcast_add(&window_mask)?;
```

Comment:
I just added them instead of using masked_fill (or where_cond), which is referred to [here](https://github.com/huggingface/candle/blob/main/candle-transformers/src/models/modernbert.rs#L226).

When values from both masks (`global_attention_mask` and `window_mask`) are `f32::MIN`, the output becomes f32::MIN + f32::MIN, and it could overflow. I honestly haven't looked into it in detail, so we need to check that it's okay.